l2geth/rollup/sync_service.go

```diff
@@ -128,22 +128,29 @@ func NewSyncService(ctx context.Context, cfg Config, txpool *core.TxPool, bc *co
 	// requirement of the remote server being up.
 	if service.enable {
 		// Ensure that the rollup client can connect to a remote server
-		// before starting.
-		err := service.ensureClient()
-		if err != nil {
-			return nil, fmt.Errorf("Rollup client unable to connect: %w", err)
+		// before starting. Retry until it can connect.
+		t1 := time.NewTicker(10 * time.Second)
+		for ; true; <-t1.C {
+			err := service.ensureClient()
+			if err != nil {
+				log.Info("Cannot connect to upstream service", "msg", err)
+			} else {
+				log.Info("Connected to upstream service")
+				t1.Stop()
+				break
+			}
 		}
 
 		// Wait until the remote service is done syncing
-		t := time.NewTicker(10 * time.Second)
-		for ; true; <-t.C {
+		t2 := time.NewTicker(10 * time.Second)
+		for ; true; <-t2.C {
```

Comment:
nit: making t1 t2 more informative names such as `t_connect` and `t_sync`?

Reply:
https://golang.org/doc/effective_go#mixed-caps